lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -297,50 +286,78 @@ export abstract class CacheManager implements ICacheManager {
                 return;
             }
 
-            if (!StringUtils.isEmpty(homeAccountId)) {
-                matches = this.matchHomeAccountId(entity, homeAccountId);
+            if (!StringUtils.isEmpty(homeAccountId) && !this.matchHomeAccountId(entity, homeAccountId)) {
+                return;
             }
 
-            if (!StringUtils.isEmpty(environment)) {
-                matches = matches && this.matchEnvironment(entity, environment);
+            if (!StringUtils.isEmpty(environment) && !this.matchEnvironment(entity, environment)) {
+                return;
             }
 
-            if (!StringUtils.isEmpty(realm)) {
-                matches = matches && this.matchRealm(entity, realm);
+            if (!StringUtils.isEmpty(realm) && !this.matchRealm(entity, realm)) {
+                return;
             }
 
-            if (!StringUtils.isEmpty(credentialType)) {
-                matches = matches && this.matchCredentialType(entity, credentialType);
+            if (!StringUtils.isEmpty(credentialType) && !this.matchCredentialType(entity, credentialType)) {
+                return;
             }
 
-            if (!StringUtils.isEmpty(clientId)) {
-                matches = matches && this.matchClientId(entity, clientId);
+            if (!StringUtils.isEmpty(clientId) && !this.matchClientId(entity, clientId)) {
+                return;
             }
 
             // idTokens do not have "target", target specific refreshTokens do exist for some types of authentication
             // TODO: Add case for target specific refresh tokens
-            if (!StringUtils.isEmpty(target) && credType === CredentialType.ACCESS_TOKEN) {
-                matches = matches && this.matchTarget(entity, target);
+            if (!StringUtils.isEmpty(target) && !this.matchTarget(entity, target)) {
```

Comment:
why did we remove `credType === CredentialType.ACCESS_TOKEN` check here? Not adding is harmless but does it makes sense to retain it?

Reply:
Never mind, I see you added the check in match Target. I still think the check should be surfaced here and have `matchTarget()` purely do its name sake operation.

Reply:
I elected to move it because it was easier to understand the logic. We can revisit this when we add target-specific refresh tokens.